src/modules/screen/consumer/screen_shader.cpp

```diff
@@ -26,37 +26,81 @@ namespace caspar { namespace screen {
 std::string get_vertex()
 {
     return R"shader(
-			#version 450
+            #version 450
             in vec4 TexCoordIn;
             in vec2 Position;
 
             out vec4 TexCoord;
-
-			void main()
-			{
-				TexCoord = TexCoordIn;
-				gl_Position = vec4(Position, 0, 1);
-			}
+	    void main()
+	    {
+		    TexCoord = TexCoordIn;
+		    gl_Position = vec4(Position, 0, 1);
+	    }
 	)shader";
 }
 
 std::string get_fragment()
 {
     return R"shader(
             #version 450
+   
+            #define COLOUR_SPACE_RGB                0
+            #define COLOUR_SPACE_DATAVIDEO_FULL     1
+            #define COLOUR_SPACE_DATAVIDEO_LIMITED  2
+
+            #define RANGE_16    (16.0f / 256.0f)
+            #define RANGE_235   (235.0f / 256.0f)
+            #define RANGE_HALF  (0.5f / 256.0f)
+
             uniform sampler2D background;
 
             in vec4 TexCoord;
             out vec4 fragColor;
 
-			uniform bool key_only;
+	    uniform bool key_only;
+            uniform int colour_space;
+            uniform int window_width;
 
-            void main() {
+            // rgb2yuv_709, rgb=0~255, y=16~235, uv=16~240
+            mat4 RGBA2YCbCrA = mat4(0.183f, -0.101f, 0.439f, 0.0f, 0.614f, -0.338f, -0.399f, 0.0f, 0.062f, 0.439f,-0.040f, 0.0f, 0.0f, 0.0f, 0.0f, 1.0f);
+   
+            vec4 dtv_color(vec4 rgba)
+            {
+              if(rgba.a > 0.0f)
+                rgba.rgb = clamp(rgba.rgb  * (1.0f / rgba.a), 0.0f, 1.0f);
+               return RGBA2YCbCrA*rgba;
+            }
+  
+            void main()
+	    {
                 vec4 color = texture(background, TexCoord.st);
-
                 if (key_only)
                     color = vec4(color.aaa, 1);
-
+                
+                else if (colour_space == COLOUR_SPACE_DATAVIDEO_FULL) {  // Full range 0-255
```

Comment:
It would be nice in this code to use color.stpq instead of rgba once color is not rgb to make it clearer. And to use TexCoord.x for the same reason